### PR TITLE
EPMRPP-36942 React. Admin part. Server settings. Statistics.

### PR DESCRIPTION
**Pull request by andrei-bulagin**

### CODE REVIEW

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+import ArrowRightIcon from 'common/img/arrow-right-inline.svg';
+import { STATISTICS_ENABLED_KEY, STATISTICS_TYPE, DEFAULT_FORM_CONFIG } from './constants';
+import styles from './statisticsTab.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  statisticsInfo: {
+    id: 'StatisticsTab.statisticsInfo',
+    defaultMessage: `You can help us improve ReportPortal by opting to send usage statistics. While you're using the app, we'll gather data that might help us improve Report Portal performance and usability by tracking usage frequency of particular features. See below for details about what information is sent.`,
```
Comment (DiscoElevator) [inline app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: Please add translations

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+          <div className={cx('image-holder')} />
+          <p className={cx('statistics-info')}>{formatMessage(messages.statisticsInfo)}</p>
+
+          <div className={cx('statistics-list', { 'list-shown': this.state.listShown })}>
+            <div className={cx('statistics-list-toggler')} onClick={this.onListClick}>
+              <span>{formatMessage(messages.statisticsList)}</span>
+              <span className={cx('arrow')}>{Parser(ArrowRightIcon)}</span>
+            </div>
+
+            <div className={cx('list-message')}>
```
Comment (DiscoElevator) [inline app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: Let's make css simpler and show/hide messages in JSX:
`{this.state.listShown && (...)}`

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    return (
+      <div className={cx('statistics-settings-tab')}>
+        <form className={cx('statistics-settings-form')} onSubmit={handleSubmit(this.onFormSubmit)}>
+          <div className={cx('image-holder')} />
+          <p className={cx('statistics-info')}>{formatMessage(messages.statisticsInfo)}</p>
+
+          <div className={cx('statistics-list', { 'list-shown': this.state.listShown })}>
+            <div className={cx('statistics-list-toggler')} onClick={this.onListClick}>
+              <span>{formatMessage(messages.statisticsList)}</span>
+              <span className={cx('arrow')}>{Parser(ArrowRightIcon)}</span>
```
Comment (DiscoElevator) [inline app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: `className={cx('arrow', {rotated: this.state.listShown})}`

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    });
+  };
+
+  onListClick = () => {
+    this.setState((state) => ({
+      listShown: !state.listShown,
+    }));
+  };
+
+  updateEmailConfig = (options) => {
```
Comment (DiscoElevator) [inline app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: 1. Email?
2. I think the only thing that сhanges here is the enabled/disabled state, so maybe it should be an arguument? Or the whole object `{ type, enabled}`. So fetch options (in this case it's `method: 'POST'`) will be incapsulated inside this function.

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    fetch(URLS.statisticsServerSettings(), options)
+      .then(() => this.updateSettingSuccess())
+      .catch(this.catchRequestError);
+  };
+
+  fetchEmailConfig = () => {
```
Comment (DiscoElevator) [inline app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: Email?

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx (hunk 1 above)
Comment (andrei-bulagin) [reply app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: Added

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx (hunk 2 above)
Comment (andrei-bulagin) [reply app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: For mobile screen it should be shown independently of this.state.listShown. This case can be implemented via css/@media only.

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx (hunk 3 above)
Comment (andrei-bulagin) [reply app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: Done

File: app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx (hunk 4 above)
Comment (andrei-bulagin) [reply app/src/pages/admin/serverSettingsPage/serverSettingsTabs/statisticsTab/statisticsTab.jsx]: Fixed, replaced to single argument

**Pull request merged**